simulation/test_web_seed.cpp

```diff
@@ -383,6 +386,118 @@ TORRENT_TEST(multi_file_redirect)
 	TEST_EQUAL(seeding, true);
 }
 
+// test web_seed redirect through proxy
+TORRENT_TEST(multi_file_redirect_through_proxy)
+{
+	using namespace libtorrent;
+	file_storage fs;
+	fs.add_file(combine_path("foo", "1"), 0xc000);
+	fs.add_file(combine_path("foo", "2"), 0xc030);
+	lt::add_torrent_params params = ::create_torrent(fs);
+	params.url_seeds.push_back("http://2.2.2.2:8080/");
+
+	bool seeding = false;
+
+	run_test(
+		[&params](lt::session& ses)
+		{
+			settings_pack pack;
+
+			pack.set_int(settings_pack::proxy_type, settings_pack::http);
+			pack.set_str(settings_pack::proxy_hostname, "50.50.50.50");
+			pack.set_str(settings_pack::proxy_username, "testuser");
+			pack.set_str(settings_pack::proxy_password, "testpass");
+			pack.set_int(settings_pack::proxy_port, 4445);
+			pack.set_bool(settings_pack::proxy_hostnames, true);
+			ses.apply_settings(pack);
```

Comment:
I see you enabled this now. has the submodule been updated to the latest version already?